apps/src/code-studio/activityUtils.js

```diff
@@ -2,34 +2,53 @@
  * A set of utility functions made for dealing with activities easier.
  */
 
+import { makeEnum } from '@cdo/apps/utils';
+
 /**
  * See ActivityConstants.
  */
 const MINIMUM_PASS_RESULT = 20;
 const MINIMUM_OPTIMAL_RESULT = 30;
 export const SUBMITTED_RESULT = 1000;
+export const LOCKED_RESULT = 1001;
```

Comment:
In this case, we should never submit with LOCKED_RESULT (such that this shouldn't end up in Activity at all I don't think). It also shouldn't end up in UserLevel table. I think this is true for SUBMITTED_RESULT as well. Instead, both of these are used to track some client side state.

I think we're kind of overloading what result means in both cases, which isn't super desirable. This is done in part because this is the only way we currently have to track user-specific data. I would love to clean some of this up in the future, but I don't think adding LOCKED_RESULT (which is fairly similar to how SUBMITTED_RESULT works) is making things significantly worse.